### ISSUE TITLE
style warnings in define-presentation-method

### ISSUE THREAD

**Issue by slyrus** [McCLIM/McCLIM#202]
When compiling forms using the define-presentation-method macro we see warnings of the following type:

```
; caught STYLE-WARNING:
;   Cannot find type for specializer
;   CLIM-INTERNALS::|(presentation-type COMMON-LISP::NIL)| when executing
;   SB-PCL:SPECIALIZER-TYPE-SPECIFIER for a STANDARD-METHOD of a
;   STANDARD-GENERIC-FUNCTION.
```

I'm wondering this isn't some sort of problem with the macro.

**Comment by dkochmanski** [comment]
could you provide a code snippet which I can copy-paste to reproduce this behavior?

using snippet below doesn't reproduce the issue:
```
(define-presentation-method presentation-type-history-for-stream
    ((type t) (stream input-editing-stream))
  ;; What is the purpose of this? Makes stuff harder to do, so
  ;; commented out...
  ;;(if (not (stream-rescanning-p stream))
  ;;       (funcall-presentation-generic-function presentation-type-history type)
  ;;       nil)
  (funcall-presentation-generic-function presentation-type-history type))
```

**Comment by slyrus** [comment]
I see the following error when building McCLIM for the first time:

```
; compiling (DEFINE-PRESENTATION-METHOD PRESENTATION-TYPEP ...)

; file: /home/sly/projects/McCLIM/Core/clim-core/presentation-defs.lisp
; in: DEFINE-PRESENTATION-METHOD PRESENTATION-TYPEP
;     (CLIM:DEFINE-PRESENTATION-METHOD CLIM:PRESENTATION-TYPEP
;                                      (CLIM-INTERNALS::OBJECT (TYPE NIL))
;                                      (DECLARE (IGNORE CLIM-INTERNALS::OBJECT))
;                                      NIL)
; --> DEFMETHOD PROGN EVAL-WHEN 
; ==>
;   (SB-PCL::%DEFMETHOD-EXPANDER CLIM-INTERNALS::%PRESENTATION-TYPEP NIL
;                                ((CLIM-INTERNALS::TYPE-KEY
;                                  CLIM-INTERNALS::|(presentation-type COMMON-LISP::NIL)|)
;                                 CLIM-INTERNALS::OBJECT TYPE)
;                                ((DECLARE (IGNORE CLIM-INTERNALS::OBJECT))
;                                 (BLOCK CLIM:PRESENTATION-TYPEP
;                                   (LET (#)
;                                     (LET #
;                                       #
;                                       #)))))
; 
; caught STYLE-WARNING:
;   Cannot find type for specializer
;   CLIM-INTERNALS::|(presentation-type COMMON-LISP::NIL)| when executing
;   SB-PCL:SPECIALIZER-TYPE-SPECIFIER for a STANDARD-METHOD of a
;   STANDARD-GENERIC-FUNCTION.
```

There are more, of course, but hopefully we can start with this one and figure out what to do, if anything, about these warnings.

**Comment by dkochmanski** [comment]
presentation types are only recorded at compilation time and it is consistent with classes.
implementations usually do not warn the user if the class is defined in the same file because they have explicit handling of such situation in their internals (i.e there is no portably solution -- clostrum would allow passing such information though), and in effect define-presentation-method which expands to defmethod has no specialization to class, because define-presentation-type does not eagerly define a class (to make it attachable to existing classes at load time).

in other words, this is wontfix.
